deployment/Deploy.ps1

```diff
@@ -3,297 +3,312 @@
+        $SubscriptionList = Get-LtiSubscriptionList
+        Write-Log -Message "List of Subscriptions:-`n$($SubscriptionList | ConvertTo-Json -Compress)"    
 
-    Write-Log -Message "Creating Managed identity inside ResourceGroup: $resourceGroupName and Identity Name: $identityName"
-    $identityObj = (az identity create -g $resourceGroupName -n $identityName) | ConvertFrom-Json
-    if(!$identityObj) {
-        throw "Encountered an Error while creating managed identity inside ResourceGroup: " + $resourceGroupName + " and Identity Name: " + $identityName
-    }
+        $SubscriptionCount = ($SubscriptionList | Measure-Object).Count
+        Write-Log -Message "Count of Subscriptions: $SubscriptionCount"
+        if ($SubscriptionCount -eq 0) {
+            throw "Please create at least ONE Subscription in your Azure Account"
+        }
+        elseif ($SubscriptionName) {
+            Write-Log -Message "Using User provided Subscription Name: $SubscriptionName"            
+        }
+        elseif ($SubscriptionCount -eq 1) {
+            $SubscriptionName = $SubscriptionList[0].name;
+            Write-Log -Message "Defaulting to Subscription Name: $SubscriptionName"
+        }
+        else {
+            $SubscriptionListOutput = $SubscriptionList | Select-Object @{ l="Subscription Name"; e={ $_.name } }, "id", "isDefault"
+            Write-Host ($SubscriptionListOutput | Out-String)
+            $SubscriptionName = Read-Host 'Enter Subscription Name from Above List'
+            Write-Log -Message "User Entered Subscription Name: $SubscriptionName"
+        }
 
-    #It takes a few seconds for the Managed Identity to spin up and be available for further processing
-    Write-Log -Message "Sleeping for 30 seconds"
-    Start-Sleep -s 30
-    Write-Host 'Managed Identity Created Successfully'
+        $ActiveSubscription = Set-LtiActiveSubscription -Name $SubscriptionName -List $SubscriptionList
+        $UserEmailAddress = $ActiveSubscription.user.name
+        #endregion
 
-    Write-Title 'STEP #7 - Creating Role Assignment'
+        #region Choose Region for Deployment
+        Write-Title "STEP #3 - Choose Location`n(Please refer to the Documentation / ReadMe on Github for the List of Supported Locations)"
 
-    $roleName = "Contributor"
-    Write-Log -Message "Assigning Role: $roleName to PrincipalID: $($identityObj.principalId)"
-    $roleAssignmentOp = az role assignment create --assignee-object-id $identityObj.principalId --assignee-principal-type ServicePrincipal --role $roleName
-    if(!$roleAssignmentOp) {
-        throw "Encountered an Error while creating Role Assignment"
-    }
+        Write-Log -Message "Fetching List of Locations"
+        $LocationList = ((az account list-locations) | ConvertFrom-Json)
+        Write-Log -Message "List of Locations:-`n$($locationList | ConvertTo-Json -Compress)"
 
-    Write-Host 'Role Assignment Created Successfully'
+        if(!$LocationName) {
```

Comment:
```suggestion
        if($LocationName) {
        Write-Log -Message "Using User provided Location Name: $LocationName"
        }else {
            Write-Host "$(az account list-locations --output table --query "[].{Name:name}" | Out-String)`n"
            $LocationName = Read-Host 'Enter Location From Above List for Resource Provisioning'
            Write-Log -Message "User Entered Location Name: $LocationName"
        }


```

Reply:
Thanks for the logging suggestion. Committing it right away.